backend_application/app/crawler/pagerank.py

```diff
@@ -69,21 +78,26 @@ def get_probabilyties(session, new_x):
     :type new_x: np.array.
     :return: matrix.
     """
-    r = get_rows(session)
+    r = rows
     M = np.empty((r, r))
     for i in range(r):
         np.set_printoptions(precision=3)
         M[:, i] = new_x[:, i] / new_x[:, i].sum()
     return M
 
-
-
-
-if __name__ == '__main__':
+def compute_pagerank():
     """
     Compute pagerank
     """
-    ses = init_db()
-    rate = dict(zip(get_urls(ses), pageRank(get_probabilyties(ses, convert_to_array(create_graph(ses))))))
-    for i, j in rate.items():
-        update_page_rank(ses, i, j)
\ No newline at end of file
+    t1 = time.time()
+    success = True
+    try:
+        ses = init_db()
+        rate = dict(zip(get_urls(ses), pageRank(get_probabilyties(get_rows(ses), convert_to_array(create_graph(ses))))))
+        for i, j in rate.items():
+            update_page_rank(ses, i, j)
+    except Exception as e:
+        success = False
+        print(f'error {type(e).__name__}: {e.args[0]}')
+    print(t1 - time.time())
```

Comment:
Time logging may cause heavy performance issues. You should move it into decorator, which would work only if DEBUG logging enabled